test/php/test_php_fallback.sh

```diff
@@ -0,0 +1,170 @@
+    local repository=$2
+    # Normally a "v" is used before the version number, but support alternate cases
+    local prefix=${3:-"tags/v"}
+    # Some repositories use "_" instead of "." for version number part separation, support that
+    local separator=${4:-"."}
+    # Some tools release versions that omit the last digit (e.g. go)
+    local last_part_optional=${5:-"false"}
+    # Some repositories may have tags that include a suffix (e.g. actions/node-versions)
+    # Try one break fix version number less if we get a failure. Use "set +e" since "set -e" can cause failures in valid scenarios.
+    set +e
+        major="$(echo "${current_version}" | grep -oE '^[0-9]+' || echo '')"
+        minor="$(echo "${current_version}" | grep -oP '^[0-9]+\.\K[0-9]+' || echo '')"
+        breakfix="$(echo "${current_version}" | grep -oP '^[0-9]+\.[0-9]+\.\K[0-9]+' 2>/dev/null || echo '')"
+
+        if [ "${minor}" = "0" ] && [ "${breakfix}" = "0" ]; then
+            ((major=major-1))
+            declare -g ${variable_name}="${major}"
+            # Look for latest version from previous major release
+            find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
+        # Handle situations like Go's odd version pattern where "0" releases omit the last part
+        elif [ "${breakfix}" = "" ] || [ "${breakfix}" = "0" ]; then
+            ((minor=minor-1))
+            declare -g ${variable_name}="${major}.${minor}"
+            # Look for latest version from previous minor release
+            find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
+        else
+            ((breakfix=breakfix-1))
+            if [ "${breakfix}" = "0" ] && [ "${last_part_optional}" = "true" ]; then
+                declare -g ${variable_name}="${major}.${minor}"
+            else 
+                declare -g ${variable_name}="${major}.${minor}.${breakfix}"
+            fi
+        fi
+    set -e
+}
+
+install_previous_version() {
+    echo -e "\nInstalling Previous Version..."
+    PHP_VERSION=$1
+    find_prev_version_from_git_tags PHP_VERSION https://github.com/php/php-src "tags/php-"
+    PHP_INSTALL_DIR="${PHP_DIR}/${PHP_VERSION}"
+    if [ -d "${PHP_INSTALL_DIR}" ]; then
+        echo "(!) PHP version ${PHP_VERSION} already exists."
+        exit 1
+    fi
+
+    if ! cat /etc/group | grep -e "^php:" > /dev/null 2>&1; then
+        groupadd -r php
+    fi
+    usermod -a -G php "${USERNAME}"
+
+    PHP_URL="https://www.php.net/distributions/php-${PHP_VERSION}.tar.gz"
+
+    PHP_INI_DIR="${PHP_INSTALL_DIR}/ini"
+    CONF_DIR="${PHP_INI_DIR}/conf.d"
+    mkdir -p "${CONF_DIR}";
+
+    PHP_EXT_DIR="${PHP_INSTALL_DIR}/extensions"
+    mkdir -p "${PHP_EXT_DIR}"
+
+    PHP_SRC_DIR="/usr/src/php"
+    mkdir -p $PHP_SRC_DIR
+    cd $PHP_SRC_DIR
+    wget -O php.tar.xz "$PHP_URL"
+    ln -sf /usr/local/php/8.1.27 /usr/local/php/current
```

Comment:
This doesn't seem right which might be causing the test failures in https://github.com/devcontainers/features/actions/runs/8256416491/job/22585068181?pr=908#step:4:26268

Also, isn't the binaries installed within `/usr/src/php`?

Reply:
I was not able to have the test show the previous version after it got installed, something to do with symlinking it seems..

Reply:
I don't understandy why there would be an issue with symlinking as all the variable are reset when we install it with previous version

https://github.com/devcontainers/features/blob/3c6f07e0f50acfc7fc0d544eaa97ef7b1d676264/src/php/install.sh#L309

Reply:
oh it was already given in code `install.sh` to resolve this. Got it working !